pipeline/terraform0.12/terraform.tf

```
@@ -0,0 +1,90 @@
+provider "aws" {
+  region = "eu-west-1"
+
+  assume_role {
+    role_arn = "arn:aws:iam::760097843905:role/platform-admin"
+  }
+}
+
+terraform {
+  required_version = ">= 0.9"
+
+  backend "s3" {
+    role_arn = "arn:aws:iam::760097843905:role/platform-developer"
+
+    bucket         = "wellcomecollection-platform-infra"
+    key            = "terraform/catalogue_pipeline_tf_0.12.tfstate"
```

```suggestion
    key            = "terraform/catalogue/pipeline.tfstate"
```

I've been trying to sort terraform states into per-repo folders in the S3 bucket, so it's a bit easier to find where any given state file is defined from. The platform-infra bucket in particular has a lot of state files now.

Also, let's not bake the terraform version into the name – that feels like a naming smell.

---

I also think this will not work as you need the state from the stack pre update.